src/components/views/LocalModList/DisableModModal.vue

```diff
@@ -1,41 +1,75 @@
 <script lang="ts">
-import { Vue, Component, Prop } from 'vue-property-decorator';
+import { Vue, Component } from 'vue-property-decorator';
 import { ModalCard } from '../../all';
+import R2Error from '../../../model/errors/R2Error';
 import ManifestV2 from '../../../model/ManifestV2';
+import LoggerProvider, { LogSeverity } from '../../../providers/ror2/logging/LoggerProvider';
+import Dependants from '../../../r2mm/mods/Dependants';
 
 @Component({
     components: {ModalCard}
 })
 export default class DisableModModal extends Vue {
 
-    @Prop({required: true})
-    readonly mod!: ManifestV2;
+    modBeingDisabled: string | null = null;
 
-    @Prop({required: true})
-    readonly dependencyList!: Set<ManifestV2>;
+    get dependants() {
+        return Dependants.getDependantList(this.mod, this.$store.state.localModList);
+    }
+
+    get isLocked(): boolean {
+        return this.modBeingDisabled !== null;
+    }
+
+    get isOpen(): boolean {
+        return this.$store.state.modals.isDisableModModalOpen
+            && this.$store.state.modals.disableModModalMod !== null;
+    }
 
-    @Prop({required: true})
-    readonly dependantsList!: Set<ManifestV2>;
+    get mod(): ManifestV2 {
+        if (this.$store.state.modals.disableModModalMod === null) {
+            this.$emit('error', new R2Error(
+                'Error while opening DisableModModal',
+                'Mod not provided'
+            ));
+        }
+        return this.$store.state.modals.disableModModalMod;
+    }
 
-    @Prop({required: true})
-    readonly modBeingDisabled!: string | null;
+    async disableModIncludingDependants() {
+        await this.disableMods([...this.dependants, this.mod]);
+    }
 
-    @Prop({required: true, type: Function})
-    readonly onClose!: () => void;
+    async disableModExcludingDependants() {
+        await this.disableMods([this.mod]);
+    }
 
-    @Prop({required: true, type: Function})
-    readonly onDisableIncludeDependents!: (mod: ManifestV2) => void;
+    private async disableMods(mods: ManifestV2[]) {
+        const onProgress = (mod: ManifestV2) => this.modBeingDisabled = mod.getName();
 
-    @Prop({required: true, type: Function})
-    readonly onDisableExcludeDependents!: (mod: ManifestV2) => void;
+        try {
+            await this.$store.dispatch(
+                'profile/disableModsFromActiveProfile',
+                { mods, onProgress }
+            );
+        } catch (e) {
+            const err: Error = e as Error;
+            LoggerProvider.instance.Log(LogSeverity.ACTION_STOPPED, `${err.name}\n-> ${err.message}`);
+            this.$emit('error', e);
+        } finally {
+            this.$store.commit('closeDisableModModal');
```

Comment:
Would it make sense for this to call `this.onClose()` instead?

Reply:
I don't know. Hopefully it does, because I changed it.